webapp/app.py

```diff
@@ -123,16 +124,31 @@ def process_invoices() -> str:
     @app.route("/process-invoices/run/<run_type>")
     @login_required
     def process_invoices_run(run_type: str) -> str | Response:
+        if run_type == "review":
+            # [TODO] Add error page for invalid run types.
+            return redirect(url_for("process_invoices_run_execute", run_type=run_type))
+        elif run_type == "final":  # noqa: RET505
+            return redirect(url_for("process_invoices_confirm_final_run"))
+        return abort(400, description=f"Invalid run type: '{run_type}'")
+
+    @app.route("/process-invoices/run/final/confirm")
+    @login_required
+    def process_invoices_confirm_final_run() -> str:
+        return render_template("process_invoices_confirm_final_run.html")
+
+    @app.route("/process-invoices/run/<run_type>/execute")
```

Comment:
**Disclaimer:** Don't think changes are needed here at this time, but wanted to comment on this.

We have a route that provides a confirmation button in the UI, which is great, but that effectively just navigates you to this route (URL).  _This_ URL does the work of starting a "final" run, then redirects the user to the status page.

The good news: most casual users will probably never know this URL route exists given the immediate redirect of their browser.

The bad news: if people _do_ know this URL exists, and they intentionally / accidentally navigate straight to it, this kind of bypasses the confirmation page behavior.

At a high level, one option could be for the confirmation button to pass some kind of secret to this route, thereby informing it that it came from a confirmation (`GET` parameter if only URL, or could change to a `POST` route and have it submit like a form, or cookies, etc., lots of options).

But to restate: I _don't_ think this is needed now, and there is a balance with how gaurd-y these confirmations need to be.  It seems unlikely to me this pattern of someone hitting this route directly, but felt worth at least observing in this PR.

Reply:
Thank you for pointing this out! I'll start looking into how we can fortify the confirmation workflow. 🤔

Reply:
@ghukill @ehanson8 Please see the proposed solution in https://github.com/MITLibraries/alma-sapinvoices-ui/pull/83/commits/1ca21d1f6a39ae182d28030c7f6e6df38145cf2c. The 'execute' route for final runs (i.e., `/process-invoices/run/final/execute` is now protected against direct get requests by using the [Flask.request.referrer](https://flask.palletsprojects.com/en/2.3.x/api/#flask.Request.referrer) to confirm that the user was redirected by the confirmation page.

Reply:
@jonavellecuerdo - awesome idea.  As mentioned in a huddle, to me this strikes a nice balance.  It provides a bit more gaurdrail in the sense you _must_ come from that button confirmation page for this to happen... but it's not quite at the level of `POST` requests + forms + secret, hidden keys getting passed around, which doesn't feel necessary.  Nice work!